src/jshint.js

```diff
@@ -4189,8 +4189,8 @@ var JSHINT = (function() {
       if (state.tokens.next.id === "var") {
         advance("var");
         state.tokens.curr.fud({ prefix: true });
-      } else if (state.tokens.next.id === "let") {
-        advance("let");
+      } else if (state.tokens.next.id === "let" || state.tokens.next.id === "const") {
```

Comment:
TODO: should only be treating let/const specially if `esnext` es3 or es5 mode is used